### PR TITLE
Add new keymap for sofle

### PR DESCRIPTION
**Pull request by axtloss**
## Description
This pr adds a new keymap for the sofle keyboard

## Types of Changes
A new keymap with some extra features and bugfixes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 1]
```diff
@@ -0,0 +1,686 @@
+#include QMK_KEYBOARD_H
```
Comment (keyboard-magpie) [inline keyboards/sofle/keymaps/noqmk/keymap.c:1]: This needs a license header (https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header) at the top.

File: keyboards/sofle/keymaps/noqmk/rules.mk [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (keyboard-magpie) [inline keyboards/sofle/keymaps/noqmk/rules.mk:1]: Extra unnecessary character at top.

File: keyboards/sofle/keymaps/noqmk/config.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+/* The way how "handedness" is decided (which half is which),
+see https://docs.qmk.fm/#/feature_split_keyboard?id=setting-handedness
+for more options.
+*/
```
Comment (keyboard-magpie) [inline keyboards/sofle/keymaps/noqmk/config.h:6]: This entire file is not currently performing any function and can be removed.

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+#include <stdio.h>
+char wpm_str[10];
```
Comment (drashna) [inline keyboards/sofle/keymaps/noqmk/keymap.c:3]:
```suggestion
```

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+            oled_write(" CLMK ", false);
+            break;
+        case _GAME:
+            oled_write(" GAME ", false);
+            break;
+        case _QWERTY:
+            oled_write("QWERT", false);
+            break;
+        default:
+            oled_write("UNDEF", false);
```
Comment (drashna) [inline keyboards/sofle/keymaps/noqmk/keymap.c:394]: Just a heads up, using the "_P" version uses less memory on the system, something of which atmega32u4's have very little of.

```suggestion
            oled_write_P(PSTR(" CLMK "), false);
            break;
        case _GAME:
            oled_write_P(PSTR(" GAME "), false);
            break;
        case _QWERTY:
            oled_write_P(PSTR("QWERT"), false);
            break;
        default:
            oled_write_P(PSTR("UNDEF"), false);
```

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 6]
```diff
@@ -0,0 +1,686 @@
...
+    }
+
+    oled_set_cursor(0, 5);
+
+    /* Print current layer */
+    oled_write("LAYER", false);
```
Comment (drashna) [inline keyboards/sofle/keymaps/noqmk/keymap.c:400]:
```suggestion
    oled_write_P(PSTR("LAYER"), false);
```

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 7]
```diff
@@ -0,0 +1,686 @@
...
+            oled_write("Raise", false);
+            break;
+        case _LOWER:
+            oled_write("Lower", false);
+            break;
+        case _ADJUST:
+            oled_write(" Adj ", false);
+            break;
+        default:
+            oled_write("Undef", false);
```
Comment (drashna) [inline keyboards/sofle/keymaps/noqmk/keymap.c:418]:
```suggestion
            oled_write_P(PSTR("Base "), false);
            break;
        case _RAISE:
            oled_write_P(PSTR("Raise"), false);
            break;
        case _LOWER:
            oled_write_P(PSTR("Lower"), false);
            break;
        case _ADJUST:
            oled_write_P(PSTR(" Adj "), false);
            break;
        default:
            oled_write_P(PSTR("Undef"), false);
```

File: keyboards/sofle/keymaps/noqmk/keymap.c [hunk 8]
```diff
@@ -0,0 +1,686 @@
...
+    }
+
+    /* caps lock */
+    oled_set_cursor(0, 8);
+    oled_write("CPSLK", led_usb_state.caps_lock);
```
Comment (drashna) [inline keyboards/sofle/keymaps/noqmk/keymap.c:423]:
```suggestion
    oled_write_P(PSTR("CPSLK"), led_usb_state.caps_lock);
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/noqmk/keymap.c (hunk 4 above)
Comment (axtloss) [reply keyboards/sofle/keymaps/noqmk/keymap.c:5]: Why exactly should these lines be removed?

Comment (daskygit) [reply keyboards/sofle/keymaps/noqmk/keymap.c:5]: `wpm_str` is not used any where in keymap.c and including stdio.h isn't necessary.

Comment (axtloss) [reply keyboards/sofle/keymaps/noqmk/keymap.c:5]: Oh alright, I'll remove it then

**Review by keyboard-magpie — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**